### PR TITLE
timings: add new helpers, Measurer interface and DurationThreshold

### PR DESCRIPTION
**Pull request by stolowski**
Add some enhancements to the timings API as a result of work on #6591:
- Measurer interface with StartSpan method, to allow passing either Timings or Span objects around.
- NewForTask helper to simplify common use case of creating Timings for a task.
- Run helper to simplify common case of running a nested measurement.
- DurationThreshold to avoid storing very short timings in the state - timings/spans that are less than that are ignored.

Some docstring cleanups were also made.

### CODE REVIEW

File: timings/helpers.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+import (
+	"github.com/snapcore/snapd/overlord/state"
+)
+
+// NewForTask creates a new Timings tree for given task.
+// Returned Timings tree has "task-id" and "change-id"
+// tags set automatically from the respective task.
+func NewForTask(task *state.Task) *Timings {
+	tags := map[string]string{"task-id": task.ID()}
```
Comment (pedronis) [inline timings/helpers.go]: I would set "task-kind" as well.

File: timings/timings.go [hunk 2]
```diff
@@ -46,6 +46,16 @@ var timeNow = func() time.Time {
 //   nestedTiming.Stop()
 //   t1.Stop()
 //   troot.Save()
+//
+// In addition, a few helpers exist to simplify typical use cases, for example the above example
+// can be reduced to:
+//   troot := timings.NewForTask(task) // tags set automatically, label and summary derived from task
```
Comment (pedronis) [inline timings/timings.go]: label and summary are not copied over anymore

File: timings/timings.go [hunk 3]
```diff
@@ -46,6 +46,16 @@ var timeNow = func() time.Time {
...
+//   t1 := troot.StartSpan("computation", "...")
+//   t1.Run("sub-computation", "...", func(nested *Span) {
+//          ... expensive computation
+//   })
+//   t1.Stop()
+//   troot.Save()
```
Comment (pedronis) [inline timings/timings.go]: missing state

File: timings/timings.go [hunk 4]
```diff
@@ -46,6 +46,16 @@ var timeNow = func() time.Time {
 //   nestedTiming.Stop()
 //   t1.Stop()
 //   troot.Save()
+//
+// In addition, a few helpers exist to simplify typical use cases, for example the above example
+// can be reduced to:
+//   troot := timings.NewForTask(task) // tags set automatically, label and summary derived from task
+//   t1 := troot.StartSpan("computation", "...")
+//   t1.Run("sub-computation", "...", func(nested *Span) {
```
Comment (pedronis) [inline timings/timings.go]: timing.Run(t1...

File: timings/state.go [hunk 5]
```diff
@@ -86,15 +95,21 @@ func flattenRecursive(data *rootTimingsJson, timings []*Span, nestLevel int, max
 // Save appends Timings data to the "timings" list in the state and purges old timings, ensuring
 // that up to MaxTimings are kept.
 // It's responsibility of the caller to lock the state before calling this function.
```
Comment (pedronis) [inline timings/state.go]: this should mention DurationThreshold as well

File: timings/state.go [hunk 6]
```diff
@@ -47,6 +47,9 @@ type rootTimingsJson struct {
 // Maximum number of timings to keep in state. It can be changed only while holding state lock.
 var MaxTimings = 100
 
+// Duration threshold - timings below the threshold will not be saved in the state.
```
Comment (pedronis) [inline timings/state.go]: same thing about lock

File: timings/helpers.go [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+ *
+ */
+
+package timings
+
+import (
+	"github.com/snapcore/snapd/overlord/state"
+)
+
+// NewForTask creates a new Timings tree for given task.
```
Comment (pedronis) [inline timings/helpers.go]: the given task

File: timings/timings.go [hunk 8]
```diff
@@ -27,14 +27,14 @@ var timeNow = func() time.Time {
 	return time.Now()
 }
 
-// Timings represents a tree of Timing measurements for a single execution of measured activity.
+// Timings represents a tree of Span measurements for a single execution of measured activity.
```
Comment (pedronis) [inline timings/timings.go]: of Span time measurements

File: timings/timings.go (hunk 4 above)
Comment (stolowski) [reply timings/timings.go]: Thanks for catching all the inconsistencies in the comments, appreciated!

**Pull request merged**